examples/import-boss/main.go

```diff
@@ -16,13 +16,15 @@ limitations under the License.
 
 // import-boss enforces import restrictions in a given repository.
 //
-// When a directory is verified, import-boss looks for a file called
-// ".import-restrictions". If this file is not found, parent directories will be
-// recursively searched.
+// When a package is verified, import-boss looks for files called
+// ".import-restrictions" in all directories between the package and
+// the $GOPATH/src.
```

Comment:
what does this mean if we become `$GOPATH`-less using go mod?

Reply:
It means we'll search for files up to the root of the filesystem.

E.g. Right now, if the program is run for `/home/user/go/src/bla.com/package`, `import-boss` will search for `.import-restrictions` up to `/home/user/go/src`, and stop there.

`$GOPATH`-less, if the program is run for `/home/user/code/my/package`, `import-boss` will search for `.import-restrictions` up to the `/home` directory.

Reply:
should we search up to a go.mod instead in non-GOPATH mode?

Reply:
Yes, we can. Should we detect if Go modules are enabled or not, so maybe search up to `go.mod` if it is, and `$GOPATH/src` if it isn't?

Reply:
would just look for the file and consider that dir as the root.

Reply:
yes, $GOPATH is the fallback.

Reply:
Okay, sounds good.

Reply:
I've implemented the search up to `go.mod`, with $GOPATH as fallback.